CRM/Core/Form.php

```diff
@@ -1302,7 +1302,11 @@ public function get($name) {
    * @return int
    */
   public function getAction() {
-    return $this->_action;
+    if (!is_numeric($this->_action)) {
+      CRM_Core_Error::deprecatedWarning('action should be an integer');
```

Comment:
I don't think this will do what you intended. `is_numeric('2')` is true. Maybe go the other way with is_string?

Reply:
@demeritcowboy no I want to quietly handle '2' - I was just worried someone might be passing in 'action=party-with-fun-name'

I wanted to add a return hint but figured unless I could be sure the fun name party was never gonna happen (ie put out some noise for a bit) I couldn't

Reply:
Ok, fun, just then on line 1307 do you want to cast to int there?

Reply:
I mean based on what you're saying and the set() function, it sounds like you would not want to cast on line 1307.

Reply:
@demeritcowboy yeah you are right - pushed